service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Pair.java

```diff
@@ -21,13 +21,6 @@ public Pair(String key, Object value) throws IllegalArgumentException
              */
             throw new IllegalArgumentException("Key cannot be null or empty");
         }
-        if (key.contains(" ") || key.contains("$") || key.contains(".") || key.length() > MAX_ALLOWABLE_KEY_LENGTH)
-        {
-            /*
```

Comment:
Please submit a separate PR on master and merge it back into preview to maintain commit consistency.

Reply:
Sure